app/assets/javascripts/float-kit/addon/services/toasts.js

```diff
@@ -0,0 +1,113 @@
+import DToastInstance from "float-kit/lib/d-toast-instance";
+import { getOwner } from "discourse-common/lib/get-owner";
+
+export default class Toasts extends Service {
+  @tracked activeToasts = new TrackedArray();
+
+  /**
+   * Render a toast
+   *
+   * @param {Object} [options] - options passed to the toast component as `@toast` argument
+   * @param {String} [options.duration] - The duration (ms) of the toast, will be closed after this time
+   * @param {ComponentClass} [options.component] - A component to render, will use `DDefaultToast` if not provided
+   * @param {String} [options.class] - A class added to the d-toast element
+   * @param {Object} [options.model] - An object which will be passed as the `@model` argument to the component
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  show(options = {}) {
+    const instance = new DToastInstance(getOwner(this), options);
+    this.activeToasts.push(instance);
+    return instance;
+  }
+
+  /**
+   * Render a DDefaultToast toast with the default theme
+   *
+   * @param {Object} [options] - @see show
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  default(options = {}) {
+    options.model.theme = "default";
+
+    return this.show({ ...options, component: DDefaultToast });
+  }
+
+  /**
+   * Render a DDefaultToast toast with the success theme
+   *
+   * @param {Object} [options] - @see show
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  success(options = {}) {
+    options.model.theme = "success";
+    options.model.icon = "check";
+
+    return this.show({ ...options, component: DDefaultToast });
+  }
+
+  /**
+   * Render a DDefaultToast toast with the error theme
+   *
+   * @param {Object} [options] - @see show
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  error(options = {}) {
+    options.model.theme = "error";
+    options.model.icon = "exclamation-triangle";
+
+    return this.show({ ...options, component: DDefaultToast });
+  }
+
+  /**
+   * Render a DDefaultToast toast with the warning theme
+   *
+   * @param {Object} [options] - @see show
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  warning(options = {}) {
+    options.model.theme = "warning";
+    options.model.icon = "exclamation-circle";
+
+    return this.show({ ...options, component: DDefaultToast });
+  }
+
+  /**
+   * Render a DDefaultToast toast with the info theme
+   *
+   * @param {Object} [options] - @see show
+   *
+   * @returns {DToastInstance} - a toast instance
+   */
+  @action
+  info(options = {}) {
+    options.model.theme = "info";
+    options.model.icon = "info-circle";
+
+    return this.show({ ...options, component: DDefaultToast });
+  }
+
+  /**
+   * Close a toast. Any object containg a valid `id` property can be used as a toast parameter.
+   */
+  @action
+  close(toast) {
+    this.activeToasts = new TrackedArray(
```

Comment:
This would be the place I mean to store history of toasts...when you close one maybe it just pushes onto a stack that is only ~10 items max, then have a button in the UI to show previous toasts. Could be a future enhancement, not saying you should do it in the UI. Just trying to avoid showing a few toasts to people and having them miss the messages